src/xbot/common/command/scripted/ScriptedCommand.java

```diff
@@ -0,0 +1,164 @@
+    }
+    
+    public ScriptedCommand(String manualScriptText, String manualScriptName, ScriptedCommandFactory availableCommandFactory) {
+        this.availableCommandFactory = availableCommandFactory;
+        
+        this.manualScriptText = manualScriptText;
+        this.manualScriptName = manualScriptName;
+        
+        log.debug("Instantiated with script string");
+    }
+    
+    private void initializeExecThread() {
+        log.debug("Initializing exec thread");
+        
+        if(scriptFile == null) {
+            log.debug("Using manual script text for exec");
+            execThread = new ScriptedCommandThread(manualScriptText, manualScriptName, this, availableCommandFactory);
+        }
+        else {
+            log.debug("Using file path for exec");
+            execThread = new ScriptedCommandThread(scriptFile, this, availableCommandFactory);
+        }
+        
+        execThread.setUncaughtExceptionHandler(new UncaughtExceptionHandler() {
+            
+            @Override
+            public void uncaughtException(Thread t, Throwable e) {
+                Logger log = Logger.getLogger(ScriptedCommandThread.class);
+                log.error("Uncaught exception in autonomous thread! " + e.toString());
+            }
+        });
+    }
+    
+    /**
+     * For internal use only!
+     * @param command
+     */
+    public synchronized void invokeCommand(Command command) {
+        log.info("Adding command " + command + " to scheduler");
+        
+        Scheduler.getInstance().add(command);
+        if(this.invokedCommands == null)
+            this.invokedCommands = new HashSet<Command>();
+        
+        this.invokedCommands.add(command);
+    }
+    
+    @Override
+    protected void initialize() {
+        initializeExecThread();
+    }
+
+    @Override
+    protected void execute() {
+        // Start the execution thread if this is the first command run
+        if(execThread.getState() == State.NEW) {
+            log.debug("Starting exec thread");
+            this.execThread.start();
+        }
+        
+        // Update any locks waiting for commands to finish
+        if(this.completionLocks != null) {
+            Set<ScriptedCommandCompletionLock> locksToRemove = new HashSet<>();
+            
+            for(ScriptedCommandCompletionLock lock : this.completionLocks) {
+                if(lock.updateLock()) {
+                    log.debug("Found lock to remove: " + lock.toString());
+                    locksToRemove.add(lock);
+                }
+            }
+            
+            if(!locksToRemove.isEmpty()) {
+                log.debug("Removing " + locksToRemove.size() + " locks");
+                completionLocks.removeAll(locksToRemove);
+            }
+        }
+    }
+
+    @Override
+    protected boolean isFinished() {
+        log.debug("isFinished called; State is " + (execThread == null ? "[uninitialized]" : execThread.getState().toString())); 
```

Comment:
Where's the other instance that I have that logic? Or are you saying that I should move that `toString` logic to a separate method in case I decide I want to log the state somewhere else?